storage/src/main/java/org/apache/uniffle/storage/handler/impl/MultiReplicaClientReadHandler.java

```diff
@@ -76,10 +76,22 @@ public ShuffleDataResult readShuffleData() {
           RssUtils.checkProcessedBlockIds(blockIdBitmap, processedBlockIds);
           return result;
         } catch (RssException e) {
-          LOG.warn(
-              "Finished read from [{}], but haven't finished read all the blocks.",
-              shuffleServerInfos.get(readHandlerIndex).getId(),
-              e);
+          // If this is not the last server, then read next.
+          if (readHandlerIndex < handlers.size() - 1) {
+            LOG.info(
+                "Finished read from {}/{} [{}], haven't finished read all the blocks. Will read from next server {}",
+                readHandlerIndex + 1,
+                handlers.size(),
+                shuffleServerInfos.get(readHandlerIndex).getId(),
+                shuffleServerInfos.get(readHandlerIndex + 1).getId());
```

Comment:
Maybe you should reserve the `e.getMessage` to help debug

Reply:
Done